app/code/community/Bolt/Boltpay/Model/Order.php

```diff
@@ -64,18 +63,15 @@ public function createOrder($reference, $sessionQuoteId = null, $isAjaxRequest =
                 throw new Exception(Mage::helper('boltpay')->__("Not all items are available in the requested quantities."));
             }
 
-            // check if the quotes matches, frontend only
-            if ( $sessionQuoteId && ($sessionQuoteId != $immutableQuote->getParentQuoteId()) ) {
-                throw new Exception(
-                    Mage::helper('boltpay')->__("The Bolt order reference does not match the current cart ID. Cart ID: [%s]  Bolt Reference: [%s]",
-                        $sessionQuoteId , $immutableQuote->getParentQuoteId())
-                );
-            }
-
-            // check if this order is currently being proccessed.  If so, throw exception
+            // check if this order is currently being processed.
             /* @var Mage_Sales_Model_Quote $parentQuote */
             $parentQuote = Mage::getModel('sales/quote')->loadByIdWithoutStore($immutableQuote->getParentQuoteId());
-            if ($parentQuote->isEmpty() ) {
+
+            /*
+             * left it here temporarily because we may have some merchants that
+             * should have it or have some problems without it.
+             **/
+            /*if ($parentQuote->isEmpty() ) {
```

Comment:
please, please don't commit commented out code. git can be used to revert removing code whenever needed. 

Can you check if you can keep this code? Given you're now creating immutable quote, parent should exist, right?

Reply:
uncommented current code